pkg/cluster/k8sres_test.go

```diff
@@ -1310,6 +1310,95 @@ func TestNodeAffinity(t *testing.T) {
 	assert.Equal(t, s.Spec.Template.Spec.Affinity.NodeAffinity, nodeAff, "cluster template has correct node affinity")
 }
 
+func TestPodAntiAffinityrRequiredDuringScheduling(t *testing.T) {
```

Comment:
could be merged with TestPodAntiAffinityPreferredDuringScheduling to become `TestGeneratePodAffinity`. Define two test cases with different configs and loop over it. That's how we do it in other places, too.